streaming/base/shared.py

```diff
@@ -268,3 +272,107 @@ def cleanup(self):
             pass
         finally:
             resource_tracker.unregister = original_rtracker_unreg
+
+
+def _pack_locals(dirnames: Set[str]) -> bytes:
+    """Pack local dirnames.
+
+    Args:
+        dirnames (Set[str]): Unpacked local dirnames.
+
+    Returns:
+        bytes: Packed local dirnames.
+    """
+    text = '\0'.join(sorted(dirnames))
+    data = text.encode('utf-8')
+    size = 4 + len(data)
+    return b''.join([np.int32(size).tobytes(), data])
+
+
+def _unpack_locals(data: bytes) -> Set[str]:
+    """Unpack local dirnames.
+
+    Args:
+        data (bytes): Packed local dirnames.
+
+    Returns:
+        Set[str]: Unpacked local dirnames.
+    """
+    size = np.frombuffer(data[:4], np.int32)[0]
+    text = data[4:size].decode('utf-8')
+    return set(text.split('\0'))
+
+
+def get_shm_prefix(my_locals: List[str], world: World) -> Tuple[str, SharedMemory]:
+    """Register or lookup our shared memory prefix.
+
+    Args:
+        my_locals (List[str]): Local working dir of each stream, relative to /. We need to verify
+            that there is no overlap with any other currently running StreamingDataset.
+        world (World): Our place in the world.
+
+    Returns:
+        Tuple[str, CreateSharedMemory]: Shared memory prefix and object. The name is required to be
+            very short for Mac OSX.
+    """
+    # Check my locals for overlap.
+    my_locals_set = set()
+    for dirname in my_locals:
+        if dirname in my_locals_set:
+            raise ValueError(f'Reused local working directory: {dirname}')
```

Comment:
Can we also extend the error message to `f'Reused local working directory: {dirname}. Provide a different one.'` ?

Reply:
done